Daemon tests fail with nil pointer panic when run twice

If you run the daemon package's tests twice in a row with `-count=2`, like so:

```
go test -count=2 -v ./internals/daemon -check.v
```

The following failure occurs (reliably as far as I can tell):

```
START: daemon_test.go:98: daemonSuite.TestAddCommand
START: daemon_test.go:61: daemonSuite.SetUpTest
PASS: daemon_test.go:61: daemonSuite.SetUpTest	0.000s

panic: runtime error: invalid memory address or nil pointer dereference
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x79f42d]

goroutine 1940 [running]:
net/http.(*onceCloseListener).close(...)
	/home/ben/sdk/go1.20.6/src/net/http/server.go:3501
sync.(*Once).doSlow(0xc00019efa8?, 0xc000169c18?)
	/home/ben/sdk/go1.20.6/src/sync/once.go:74 +0xc2
sync.(*Once).Do(...)
	/home/ben/sdk/go1.20.6/src/sync/once.go:65
net/http.(*onceCloseListener).Close(0xc000223920?)
	/home/ben/sdk/go1.20.6/src/net/http/server.go:3497 +0x4a
panic({0x936020, 0xd7d7a0})
	/home/ben/sdk/go1.20.6/src/runtime/panic.go:884 +0x213
net/http.(*onceCloseListener).Accept(0xa88ef8?)
	<autogenerated>:1 +0x24
net/http.(*Server).Serve(0xc00019ef00, {0x0, 0x0})
	/home/ben/sdk/go1.20.6/src/net/http/server.go:3059 +0x385
github.com/canonical/pebble/internals/daemon.(*Daemon).Start.func1()
	/home/ben/w/pebble/internals/daemon/daemon.go:490 +0x93
gopkg.in/tomb%2ev2.(*Tomb).run(0xc0004a46f8, 0xc0003ec600?)
	/home/ben/go/pkg/mod/gopkg.in/tomb.v2@v2.0.0-20161208151619-d5d1b5820637/tomb.go:163 +0x36
created by gopkg.in/tomb%2ev2.(*Tomb).Go
	/home/ben/go/pkg/mod/gopkg.in/tomb.v2@v2.0.0-20161208151619-d5d1b5820637/tomb.go:159 +0xee
FAIL	github.com/canonical/pebble/internals/daemon	10.135s
```

It probably means we still have something wrong with our Close logic (see also https://github.com/canonical/pebble/pull/253).

I've spent enough time on fixing test failures today, so opening this to track if someone has some time on a rainy day.

--- a/internals/daemon/daemon_test.go
+++ b/internals/daemon/daemon_test.go
@@ -69,6 +69,7 @@ func (s *daemonSuite) SetUpTest(c *C) {
 		c.Fatalf("cannot start reaper: %v", err)
 	}
 
+	s.socketPath = ""
 	s.pebbleDir = c.MkDir()
 	s.statePath = filepath.Join(s.pebbleDir, cmd.StateFile)
 	systemdSdNotify = func(notif string) error {

